src/JSInterop/Microsoft.JSInterop/src/Microsoft.JSInterop.WarningSuppressions.xml

```diff
@@ -7,6 +7,12 @@
       <property name="Scope">member</property>
       <property name="Target">M:Microsoft.JSInterop.Infrastructure.DotNetDispatcher.ParseArguments(Microsoft.JSInterop.JSRuntime,System.String,System.String,System.Type[])</property>
     </attribute>
+    <attribute fullname="System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessageAttribute">
+      <argument>ILLink</argument>
+      <argument>IL2065</argument>
+      <property name="Scope">member</property>
+      <property name="Target">M:Microsoft.JSInterop.Infrastructure.DotNetDispatcher.ScanAssemblyForCallableMethods(Microsoft.JSInterop.Infrastructure.DotNetDispatcher.AssemblyKey)</property>
```

Comment:
I don't understand why this is being suppressed in the XML file here, but the method currently has 2 suppressions in attributes.

```C#
        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", Justification = "We expect application code is configured to ensure JSInvokable methods are retained. https://github.com/dotnet/aspnetcore/issues/29946")]
        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2072", Justification = "We expect application code is configured to ensure JSInvokable methods are retained. https://github.com/dotnet/aspnetcore/issues/29946")]
        private static Dictionary<string, (MethodInfo, Type[])> ScanAssemblyForCallableMethods(AssemblyKey assemblyKey)
```

Shouldn't the `IL2072` in the attribute be changed to `IL2065` and this entry removed?

Reply:
I'll do a follow up to remove the attributes. It's more consistent to have all of the suppressions in xml (particularly since the attributes are unable to be applied to generated code).

Reply:
> It's more consistent to have all of the suppressions in xml 

It's less about consistency, and more about what warnings a user sees. Warnings that are suppressed in attributes in the code will always suppress the warning - the user will never see it. Warnings that are suppressed in the xml file are only suppressed for the build of ASP.NET. A user trimming their application will still see the warning (which in most cases they can't do anything about).

Reply:
That makes sense. In this particular case, users really don't use these APIs (the type is internal and buried deep in our stack) so either option is just as effective with the xml being preferable because it puts it all in one place.